### PR TITLE
Adding documentation explaining app service implementation

### PR DESCRIPTION
**Pull request by gpcastro**

### CODE REVIEW

File: doc/hosts/appservice.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+in a single file, `node_modules.tar.gz`, that is located in `/home/site/wwwroot`. As a result, we no longer maintain the
+`node_modules` folder in this `/home/site/wwwroot`.
+
+As part of the application startup process, we extract the contents of `node_modules.tar.gz` to `/node_modules`, which
+is outside of the volume share. Since the node runtime looks for packages inside directories called `node_modules`
+starting at the application directory (`/home/site/wwwroot`) all the way to `/`, it is able to find the extracted
+packages.
+
+If for some reason you want to disable this behavior, e.g. you have hardcoded references to files inside 
+`node_modules`, you can set the app setting `BUILD_FLAGS` to `Off`. Note that casing matters, so `off` won't work.
```
Comment (kichalla) [inline doc/hosts/appservice.md]: Interesting, so `BUILD_FLAGS` is the flag that AppService chose to indicate this? how clear :-)

File: doc/hosts/appservice.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Oryx is still invoked to detect how to start an application if no start up command was specified.
+
+Here we describe some details of this process, and how you might configure them and fix issues
+if needed. We focus on the specifics for App Service; for how we support each language/runtime in
+general, please refer to their specific entry in our [docs page](../README.md).
+
+[Azure App Service on Linux]: https://docs.microsoft.com/en-us/azure/app-service/containers/app-service-linux-intro
+[continuous deployment]: https://docs.microsoft.com/en-us/azure/app-service/deploy-continuous-deployment?toc=%2fazure%2fapp-service%2fcontainers%2ftoc.json
+
+## Overview
```
Comment (kichalla) [inline doc/hosts/appservice.md]: 👏

File: doc/hosts/appservice.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+designed for [storage][], which might also include backups, replication, and much more.
+
+[Kudu]: https://github.com/azure-App-Service/kudulite
+[storage]: https://azure.microsoft.com/en-us/product-categories/storage/
+
+## Node.js
+
+In general, node.js applications have a large number of package dependencies, either directly or indirectly,
+i.e. the dependencies of their dependencies. Since each package might contain several `.js` files, fetching 
+dependencies means a lot of disk IO operations. Since in the App Service model the application is stored in a
```
Comment (kichalla) [inline doc/hosts/appservice.md]: nit: IO => I/O

File: doc/hosts/appservice.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+persistent storage location, some web apps use it to store state. We don't recomment this since
+it's easy to run into concurrent access issues. We recommend the use of services specifically
+designed for [storage][], which might also include backups, replication, and much more.
+
+[Kudu]: https://github.com/azure-App-Service/kudulite
+[storage]: https://azure.microsoft.com/en-us/product-categories/storage/
+
+## Node.js
+
+In general, node.js applications have a large number of package dependencies, either directly or indirectly,
```
Comment (dorfire) [inline doc/hosts/appservice.md]: Node.js (capital N)

File: doc/hosts/appservice.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+i.e. the dependencies of their dependencies. Since each package might contain several `.js` files, fetching
```
Comment (dorfire) [inline doc/hosts/appservice.md]: using `i.e.` is discouraged in the MS style guide: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/i/ie

File: doc/hosts/appservice.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+## .NET Core
+
+The .NET Core DLL files are located in `/home/site/wwwroot`, and since this is a shared location, concurrent access
+issues might happen. For instance, if you updated your application and don't see your changes reflected when you
+use it, set the application setting `APP_SVC_RUN_FROM_COPY` to `true`. This will make the application run from a
+location other than `/home/site/wwwroot`. 
+
+When using this solution, you should not have hardcoded references to files that include `/home/site/wwwroot` in 
+the path, e.g. in custom startup scripts. Alternatively, you can just reference those files by its relative path,
```
Comment (dorfire) [inline doc/hosts/appservice.md]: same for e.g.:
https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg

File: build/build-buildimages.sh [hunk 7]
```diff
@@ -13,17 +13,18 @@ source $REPO_DIR/build/__variables.sh
 
 cd "$BUILD_IMAGES_BUILD_CONTEXT_DIR"
 
-declare BUILDSCRIPT_SOURCE="buildscriptbuilder"
 declare BUILD_SIGNED=""
 
+echo "SignType is: "$SIGNTYPE
```
Comment (dorfire) [inline build/build-buildimages.sh]: where did this come from?

**Pull request merged**